### PR TITLE
[Proposal] anva-kn initial proposal

### PR DESCRIPTION
**Pull request by anva-kn**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [ ] passing tests/CI

This PR is an initial proposal for the term project. Reference sections still need to be updated.

### CODE REVIEW

File: proposal/anva-kn/README.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Nonetheless, they are GUI programs that cost money and don't integrate the pipelines that data scientists and ML researchers have established. What that community need is a python library.
+
+# Prospective Users
+
+Data scientists, scientists, and enthusiasts that are willing to explore spectroscopic data, settle for nothing less than full control of their program, don't mind writing code for that, and most importantly, do not wish to pay!
+
+# System Architecture
+
+The module should be compatible with NumPy arrays
```
Comment (yungyuc) [inline proposal/anva-kn/README.rst:22]: This is a good feature to have for processing a lot of data.

File: proposal/anva-kn/README.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+![ramanflow_flowchart.jpeg](../_resources/ramanflow_flowchart.jpeg)
```
Comment (yungyuc) [inline proposal/anva-kn/README.rst:24]: The picture doesn't seem to be available.

File: proposal/anva-kn/README.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# API Description
+
+Ideally, it would be available on PyPI, and one can install it using pip.
+
+However, an alternative means of installation would be git cloning the repo and installing using make.
```
Comment (yungyuc) [inline proposal/anva-kn/README.rst:30]: Github is a better way to recruit collaborators.

File: proposal/anva-kn/README.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+- Read/Save the data
+    - Read .txt format files
+    - Read .dat format files
+    - Read .tiff format files
+    - Read all files (.txt, .dat, tiff) inside one directory
+- Preprocess the data
+    - Remove noise
+        - FFT noise removal (TBD)
```
Comment (yungyuc) [inline proposal/anva-kn/README.rst:39]: Do you plan to implement FFT yourself?

File: proposal/anva-kn/README.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+- Version Control
+
+- Git
+
+# Schedule
+
+| Week | To-Do |
+| --- | --- |
+| 1   | Implement read module |
+| 2   | Implement Preprocessing part 1 |
```
Comment (yungyuc) [inline proposal/anva-kn/README.rst:87]: What are included in parts 1 and 2 in both preprocessing and processing?

**Review by yungyuc — CHANGES_REQUESTED** [review]
Look like a clear plan.  It's better to resolve some of the points.

**Comment by anva-kn** [comment]
Dear professor. I've cleaned up the pr commit history. Thank you very much for taking your time to review it!

File: proposal/anva-kn/README.rst (hunk 2 above)
Comment (yungyuc) [reply proposal/anva-kn/README.rst]: Now it shows.

File: proposal/anva-kn/README.rst (hunk 4 above)
Comment (yungyuc) [reply proposal/anva-kn/README.rst:46]: Sounds good to me.

File: proposal/anva-kn/README.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
++-------+-------------------------------------------------------------------------+
+| 2     |   Implement Preprocessing part 1 (Noise removal)                        |
++-------+-------------------------------------------------------------------------+
+| 3     |   Implement Preprocessing part 2 (Cosmic rays removal, SNR)             |
++-------+-------------------------------------------------------------------------+
+| 4     |   Implement Processing part 1 (Background modeling and interpolation)   |
++-------+-------------------------------------------------------------------------+
+| 5     |   Implement Processing part 2 (Peak fitting)                            |
++-------+-------------------------------------------------------------------------+
+| 6     |   Binding to python, unittesting, debugging, writing basic documentation|
```
Comment (yungyuc) [inline proposal/anva-kn/README.rst:103]: Unit tests are usually add along with the first working functions.  The later you add unit tests, the more time it takes you to finish a project.

You do not need to write tests before features as TDD suggested, though.  In my experience TDD is usually counter-productive.

**Review by yungyuc — APPROVED** [review]

**Pull request merged**